pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -217,3 +230,66 @@ func InjectSpanID(executionContext *ExecutionStartInfo, headers http.Header) {
 		executionContext.SpanID = value
 	}
 }
+
+func capturePayloadAsTags(value interface{}, targetSpan *pb.Span, key string, depth int, maxDepth int) {
+	if key == "" {
+		return
+	}
+	if value == nil {
+		targetSpan.Meta[key] = ""
+		return
+	}
+	if depth >= maxDepth {
+		switch value := value.(type) {
+		case map[string]interface{}:
+			targetSpan.Meta[key] = convertJSONToString(value)
+		default:
+			targetSpan.Meta[key] = fmt.Sprintf("%v", value)
+		}
+		return
+	}
+	switch value := value.(type) {
```

Comment:
so this switch covers the 1st pass? then when depth is incremented i.e. greater than 0 then the above switch takes over?

Reply:
you are overthinking this. this function is agnostic on which depth it is at. the "above switch" is just try to format JSON string better.